### PR TITLE
[1/Z] Add use of datastore for storing and retrieving comments.

### PR DESCRIPTION
**Pull request by nikhitashanker**
Currently, we store comments an ArrayList such that when the server restarts, all data is lost. In this change, we store comments in the data store and retrieve comments from the datastore to acheive persistent storage.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,10 +33,24 @@
...
-  private List<Comment> comments = new ArrayList<Comment>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
```
Comment (hlz03) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Can be a separate private function e.g. buildComments

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -42,9 +62,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String text = getParameter(request, "comment-text", "");
     String commenterName = getParameter(request, "commenter-name", "Anonymous");
     String commenterEmail = getParameter(request, "commenter-email", "Unknown");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("commenterEmail", commenterEmail);
+    commentEntity.setProperty("commenterName", commenterName);
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("timestamp", timestamp);
```
Comment (hlz03) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: can be a separate function.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -42,9 +56,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String text = getParameter(request, "comment-text", "");
     String commenterName = getParameter(request, "commenter-name", "Anonymous");
     String commenterEmail = getParameter(request, "commenter-email", "Unknown");
+    long timestamp = System.currentTimeMillis();
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: nit: I think you can inline 'timestamp'.

Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

**Pull request merged**